url9ecf1e8297_nhamidi_ninja_tgz-pJ8-net_fec_openrq_test_TestRunnerJ8/src/RecepteurThreadMulti.java

```diff
@@ -0,0 +1,176 @@
+	this.port = port;
+	this.endLoop = true;
+	this.needMore = false;
+	this.socketReception = new MulticastSocket(port);
+	this.socketReception.joinGroup(groupeIP);
+	this.numberOfPackets = 0;
+	this.endReception = 1;
+	this.packetData = 0;
+	this.nbOfReceiver = nbOfRecv;
+	this.totalSymb = 1;
+	this.time_1 = 0;
+	this.time_2 = 0;
+	this.nbAck = 0;
+	this.historic = historic;
+	for (int i = 0; i < tab_lost.length; i++)
+	    this.tab_lost[i] = 0;
+	
+    }
+    
+    public boolean getNeedMore() {
+	return this.needMore;
+    }
+    
+    public void setNeedMore(boolean var) {
+	this.needMore = var;
+    }
+    
+    public boolean getStatusEndLoop() {
+	return this.endLoop;
+    }
+    
+    public void changeStatusEndLoop() {
+	this.endLoop = false;
+    }
+    
+    public void setTotalNumberOfPackets(int nb_packet) {
+	this.totalSymb = nb_packet;
+    }
+    
+    public int getNumberOfPackets() {
+	return this.numberOfPackets;
+    }
+    
+    public void setNumberOfPackets(int nb_packets) {
+	this.needMore = true;
+	if ( this.numberOfPackets <= nb_packets ) {
+	    
+	    if ( this.historic ) {
+		
+		for (int i = 0; i < tab_lost.length; i++) {
+		    if ( tab_lost[i] == 0.0 ) {
+			tab_lost[i] = ((float) nb_packets / (float) this.totalSymb);
+			break;
+		    }
+		}
+		
+		System.out.println("             " + nb_packets + "       " + this.totalSymb);
+		System.out.println("      Plost                " + ((float) nb_packets / (float) this.totalSymb) + "            " + tab_lost[0]
+			+ "            " + tab_lost[1] + "            " + tab_lost[2]);
+		int packet_sup = 10;
+		this.numberOfPackets = (int) Math.round((((float) nb_packets / (float) this.totalSymb) + 1) * (float) nb_packets) + packet_sup;
+		
+	    } else
+		this.numberOfPackets = nb_packets;
+	    
+	    setTotalNumberOfPackets(this.numberOfPackets);
+	    
+	}
+	
+    }
+    
+    public void resetNumberOfPackets() {
+	this.numberOfPackets = 0;
+    }
+    
+    public long getTime_1() {
+	return this.time_1;
+    }
+    
+    public long getTime_2() {
+	return this.time_2;
+    }
+    
+    public int getNbAck() {
+	return this.nbAck;
+    }
+    
+    public void setTimeSimu() {
+	this.before = System.currentTimeMillis();
+    }
+    
+    public void run() {
+	
+	while (true) {
+	    byte[] receiveData = new byte[50];
+	    
+	    DatagramPacket receivePacket = new DatagramPacket(receiveData, receiveData.length);
+	    
+	    try {
+		socketReception.receive(receivePacket);
+		byte[] packetData_2 = receivePacket.getData();
+		for (int l = 0; l < Utils.IntegerSize; l++) {
+		    this.packetData = (this.packetData << 8) + (packetData_2[l] & 0xff);
+		}
+		this.needMore = false;
+	    } catch (Exception exc) {
+		System.out.println(exc);
+	    }
+	    System.out.println("On a recu :                " + this.packetData);
+	    this.nbAck++;
+	    
+	    try {
+		Thread.sleep(Utils.RTT);
+	    } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)